fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java

```diff
@@ -416,7 +416,7 @@ public void analyze(Analyzer analyzer) throws UserException {
             if (distributionDesc == null) {
                 throw new AnalysisException("Create olap table should contain distribution desc");
             }
-            distributionDesc.analyze(columnSet);
+            distributionDesc.analyze(columnSet, columnDefs);
```

Comment:
There is another place we need to modify in this `CreateTableStmt`.
When user create a DUPLICATE KEY table without specify the key column, Doris will auto
select part of columns as KEY column.
And if it select a string column as KEY column, the error will return to user.
But this is not what user expected.
We should:
1. stop selecting KEY column when met String column.
2. return error msg to user when the first column is String and give a suggestion to user.